public/editor-client/src/api/index.ts

```diff
@@ -1206,3 +1217,163 @@ export const updateGlobalBlocks = async (
 };
 
 //#endregion
+
+//#region Default Templates
+
+export const getDefaultKits = async (
+  url: string,
+  id: string
+): Promise<{
+  blocks: Kit[];
+  categories: { slug: string; title: string }[];
+  styles: Style;
+}> => {
+  const fullUrl = makeUrl(url, {
+    project_id: id
+  });
+
+  const response = await request(fullUrl, {
+    method: "GET"
+  });
+
+  if (response.ok) {
+    const res = await response.json();
+
+    return {
+      blocks: res.collections,
+      categories: res.categories,
+      styles: res.styles
+    };
+  }
+
+  throw new Error(t("Failed to load kits"));
+};
+
+export const getKitData = async (
+  url: string,
+  kitId: string,
+  id: string
+): Promise<DefaultBlock> => {
+  const fullUrl = makeUrl(url, {
+    project_id: kitId,
+    page_slug: id
+  });
+
+  const response = await request(fullUrl, {
+    method: "GET"
+  });
+
+  if (response.ok) {
+    const res = await response.json();
+    const collection = res.collection.pop();
+
+    return JSON.parse(collection.pageData).items.pop();
```

Comment:
use json reader of wrap your JSON.parse in try/catch + after parsing check if your result is array before using `.pop` method